### DISCUSSION TITLE
[questions] Q1.  possible to use limiter: { max: 10, duration: 1000 } & Manual rate-limit  (RateLimitError) together? Q2: how to set Manual rate-limit by using @nestjs/bullmq

### DISCUSSION THREAD

**Discussion by grimmerk** [taskforcesh/bullmq#1949] [Q&A]
Q1:  There are two scenarios about rate limits in my project. I wonder if is it possible to use apply them together. 

Q2: 
I do not find out how to set a Manual rate-limit in `@nestjs/bullmq` framework. Is it possible?

**Comment by manast** [comment]
Regarding Q1: https://blog.taskforce.sh/rate-limit-recipes-in-nodejs-using-bullmq/

Regarding Q2, please check in @nestjs repo.

**Reply by grimmerk** [reply]
@manast  thank you for the kind answers. 

Regarding Q1, there is no clear description to say the dynamic rate limiter and the basic rate-limiter can work at the same time. I saw the below code in the Dynamic rate-limit section in the blog. Therefore, I wanna to double confirm with you for this detail. Does it mean that they can work together, or the dynamic rate limiter will deprecate the basic one, ` limiter: { max: 300, duration: 1000 },`?
 
```
const rateLimitWorker = new Worker(
  "rate-limit",
  async (job) => {
    const result = await callExternalAPI(job.data);
    if (rateLimited(result)) {
      await rateLimitWorker.rateLimit(1000);
      throw Worker.RateLimitError();
    }
  },
  {
    connection: { host: "my.redis-host.com" },
    concurrency: 500,
    limiter: { max: 300, duration: 1000 },
    attempts: 5,
    backoff: {
      type: "exponential",
      delay: 1000,
    },
  }
);
```

**Reply by manast** [reply]
They can work together with the manual rate limiter overriding the global one. The best is if you write some tests to see how it behaves in practice.
